### ISSUE TITLE
`Tabs` component crashes in React.StrictMode

### ISSUE THREAD

**Issue by eysterbrian** [zendeskgarden/react-components#1407]
Similar to the issue with `ButtonGroup` https://github.com/zendeskgarden/react-components/issues/1162, the `Tabs` component crashes when run in React's StrictMode (with React v17 or v18).

## Expectations

The `Tabs` component should allow different tabs to be selected, when run using `<React.StrictMode>`.

## Reality

But when built using `React.StrictMode` (and run in a dev build) clicking any tabs yields the error: "Cannot read properties of null (reading 'focus')"

<img width="468" alt="image" src="https://user-images.githubusercontent.com/48030216/189700889-207af577-de5c-47ed-b001-0b753b109469.png">

## Steps to Reproduce

1. Here's a fork of the [`Tabs` "Default" docs example](https://garden.zendesk.com/components/tabs#default), with the only change being that the component is wrapped in `<React.StrictMode>` in `index.tsx`: https://codesandbox.io/s/peaceful-proskuriakova-40cvdr

### Fine Print

- Component: `Tabs`
- Version: latest version (8.55)  
- Browsers: all

**Comment by melanieseltzer** [comment]
It's likely because of React 18's [new dev-only Strict Mode behaviors](https://reactjs.org/blog/2022/03/29/react-v18.html#new-strict-mode-behaviors) RE: unmounting/remounting. 

I was curious and did a little debugging with some logs, am getting different results for [these array lengths](https://github.com/zendeskgarden/react-containers/blob/main/packages/selection/src/useSelection.ts#L25-L26) in Strict Mode vs. non Strict Mode.

**Non strict mode result:**
<img width="398" alt="Screen Shot 2022-09-12 at 5 55 53 PM" src="https://user-images.githubusercontent.com/17421347/189784166-41b7fa68-03b6-4353-92fe-b2d015874bf1.png">

**Strict mode result:**
<img width="407" alt="Screen Shot 2022-09-12 at 5 42 32 PM" src="https://user-images.githubusercontent.com/17421347/189783525-f649bfcc-3aa3-4532-8248-231de05abe32.png">

I guess with the new remounting behavior in Strict Mode, [the arrays are getting populated twice](https://github.com/zendeskgarden/react-containers/blob/main/packages/selection/src/useSelection.ts#L78-L79) and the syncing between them is off, with [this](https://github.com/zendeskgarden/react-containers/blob/main/packages/selection/src/useSelection.ts#L38) no longer being a reliable way to find the focused index:

```js
const focusedIndex = items.indexOf(controlledFocusedItem);
refs[focusedIndex] && refs[focusedIndex].current!.focus(); // `current` could be null
```

Not 100% sure of a solution, but just thought this breakdown might be helpful 🙂

**Comment by eysterbrian** [comment]
Thanks for looking into this, @melanieseltzer.  

I wondered, too, about that unmounting/remounting behavior in React 18 StrictMode.  But I did find that this problem exists with React v17 as well as v18 fwiw.

**Comment by Francois-Esquire** [comment]
Hey @eysterbrian thanks for bringing this up, and @melanieseltzer thank you for looking more into this. We have had plans to address the React 18 migration, however it's not at the top of our priority list at the moment - it is on the list though, and we will be addressing this soon!

**Comment by jzempel** [comment]
Fixed by #1580
